stepped-config/src/main.js

```diff
@@ -0,0 +1,137 @@
+var cc = DataStudioApp.createCommunityConnector();
+
+// https://developers.google.com/datastudio/connector/reference#isadminuser
+function isAdminUser() {
+  return false;
+}
+
+function optionsForState(state) {
+  switch (state) {
+    case 'IL': {
+      return [['Chicago', 'CHI'], ['Springfield', 'SPI']];
+    }
+    case 'CA': {
+      return [['Mountain View', 'MTV'], ['San Francisco', 'SFO']];
+    }
+    default: {
+      cc.newUserError()
+        .setText('You must either select "CA" or "IL"')
+        .throwException();
+    }
+  }
+}
+
+function getConfig(request) {
+  var configParams = request.configParams;
+  var isFirstRequest = configParams === undefined;
+  var config = cc.getConfig();
+
+  config
+    .newSelectSingle()
+    .setId('state')
+    .setName('State')
+    .setIsDynamic(true)
+    .addOption(
+      config
+        .newOptionBuilder()
+        .setLabel('Illinois')
+        .setValue('IL')
+    )
+    .addOption(
+      config
+        .newOptionBuilder()
+        .setLabel('California')
+        .setValue('CA')
+    );
+
+  if (isFirstRequest) {
```

Comment:
Does it make sense to add a comment here to explain what isFirstRequest is?

Relevant: I feel places where stepped config related things appear, there should be a comment with a link or note explaining what the thing is for.

Reply:
Good point. Add in clarification where relevant.